src/Kunstmaan/AdminBundle/Helper/Security/Acl/AclNativeHelper.php

```diff
@@ -45,20 +50,126 @@ class AclNativeHelper
     public function __construct(EntityManager $em, TokenStorageInterface $tokenStorage, RoleHierarchyInterface $rh, $permissionsEnabled = true)
     {
         $this->em = $em;
+        $this->databasePlatform = $this->em->getConnection()->getDatabasePlatform()->getName();
         $this->tokenStorage = $tokenStorage;
         $this->roleHierarchy = $rh;
         $this->permissionsEnabled = $permissionsEnabled;
     }
 
     /**
-     * Apply the ACL constraints to the specified query builder, using the permission definition
+     * Apply the ACL constraints to the specified query builder, using the permission definition, for all database platforms
      *
      * @param QueryBuilder         $queryBuilder  The query builder
      * @param PermissionDefinition $permissionDef The permission definition
      *
      * @return QueryBuilder
      */
     public function apply(QueryBuilder $queryBuilder, PermissionDefinition $permissionDef)
+    {
+        if ($this->databasePlatform == 'postgresql') {
```

Comment:
See comments in `AclHelper.php` about duplicate query

Reply:
Same answer as above applies. The original 2811 was tested against one of KBC core system and caused crashes all over the place. To make abolutely sure backward compatibility with msql is guearanted I opted to do a separate method for postgres. This is not critical/essential however and could be refactored to a single method with some mor if(){}else{} clauses. Apart from processing a few more ifs, IMO this makes the code less readable, so I would not do it ifg the product was my own. However I also see that this can be a matter of taste.